dev/com.ibm.ws.jaxrs.2.0.cdi.1.2_fat/test-applications/contextandCDI/src/com/ibm/ws/jaxrs20/cdi12/fat/contextandCDI/CDIFilter.java

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.jaxrs20.cdi12.fat.contextandCDI;
+
+import java.io.IOException;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.servlet.ServletContext;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.core.Context;
+
+@ApplicationScoped
+public class CDIFilter implements ContainerRequestFilter, ContainerResponseFilter {
+
+    @Context ServletContext servletContext;
```

Comment:
Just a thought here... I think it is also possible for CDI to inject the ServletContext (using `@Inject ServletContext servletContext2;` - maybe we could include this in the test too?  That would ensure that ServletContext injection works regardless of who injects it.